cpp/coefficients.cpp

```diff
@@ -5,19 +5,54 @@
 // SPDX-License-Identifier:    MIT
 
 #include "coefficients.h"
-#include "QuadratureRule.h"
 #include "error_handling.h"
 #include "geometric_quantities.h"
 #include <basix/quadrature.h>
 #include <dolfinx/mesh/cell_types.h>
 using namespace dolfinx_contact;
 
+void dolfinx_contact::transformed_push_forward(
+    const dolfinx::fem::FiniteElement* element, cmdspan4_t reference_basis,
+    std::vector<double>& element_basisb, mdspan3_t basis_values, mdspan2_t J,
+    mdspan2_t K, double detJ, std::size_t basis_offset, std::size_t q,
+    std::int32_t cell, std::span<const std::uint32_t> cell_info)
+{
+  const std::function<void(const std::span<PetscScalar>&,
+                           const std::span<const std::uint32_t>&, std::int32_t,
+                           int)>
+      transformation = element->get_dof_transformation_function<PetscScalar>();
+  // Get push forward function
+  using xu_t = stdex::mdspan<double, stdex::dextents<std::size_t, 2>>;
```

Comment:
I think we can use `dolfinx_contact::mdspan2_t` and `dolfinx_contact::cmdspan2_t` for these